test/unit/services/service_creation_service_test.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class ServiceCreationServiceTest < ActiveSupport::TestCase
+  setup do
+    @provider = FactoryGirl.create(:simple_provider)
+  end
+
+  test 'call' do
+    assert_difference '@provider.services.count' do
+      ServiceCreationService.call(@provider, name: 'my-api', system_name: 'my-api')
+    end
+  end
+
+  test 'kubernetes_service_link' do
+    service_creation = ServiceCreationService.call(@provider, name: 'my-api', system_name: 'my-api', kubernetes_service_link: '/api/v1/namespaces/fake-project/services/fake-api')
+    assert_equal '/api/v1/namespaces/fake-project/services/fake-api', service_creation.service.kubernetes_service_link
+  end
+
+  test 'async' do
+    service_attributes = { name: 'my-api', namespace: 'my-project' }
+    assert_no_difference '@provider.services.count' do
+      ServiceDiscovery::ImportClusterServiceDefinitionsWorker.expects(:perform_async).with(@provider.id, *service_attributes.values_at(:namespace, :name))
+      ServiceCreationService.call(@provider, service_attributes.merge(source: 'discover'))
```

Comment:
It looks like `source: 'discover'`  is important though there is no test for when it is absent

Reply:
Done